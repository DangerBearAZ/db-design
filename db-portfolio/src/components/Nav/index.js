import React from 'react';

function Nav() {

    const categories = [
        {
            name: "Projects",
            description: "Things I have made.",
        },
        { name: "Resume", description: "My Resume." },
    ];


    function categorySelected() {
        console.log(`clicked`)
    }

    return (
        <header className="flex-row px-1">
            <h2>
                <a data-testid="link" href="/">
                    <span role="img" aria-label="skull">ðŸ’€</span> Danger By Design
                </a>
            </h2>
            <nav>
                <ul className="felx-row">
                    <li className="mx-2">
                        <a data-testid="about" href="#about">
                            About Me
                        </a>
                    </li>
                    <li>
                        <span>Contact</span>
                    </li>
                    {categories.map((category) => (
                        <li
                            className="mx-1"
                            key={category.name}
                        >
                             <span onClick={() => categorySelected(category.name)} >
                                {category.name}
                            </span>
                        </li>
                    ))}
                </ul>
            </nav>
        </header>
    );
}

export default Nav;
