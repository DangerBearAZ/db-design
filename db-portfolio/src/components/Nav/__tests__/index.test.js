import React from 'react';
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect';
import Nav from '..'; 

afterEach(cleanup);

describe('Nav component', () => {
    //baseline 
    it('renders', () => {
        render(<Nav />);
    });

    //snapshot 
    it('matches snapshot', () => {
        const { asFragment } = render(<Nav />);
        expect(asFragment()).toMatchSnapshot();
    });
});

// check for emoji 
describe('emoji is visible', () => {
    it('inserts emoji into the h2', () => {
        const { getByLabelText } = render ( <Nav /> );

        expect(getByLabelText('skull')).toHaveTextContent('ðŸ’€');
    });
});

//test for links 
describe('links are visable', () => {
    it('inserts text into the links', () => {
        //arrange 
        const { getByTestId } = render(<Nav />);
        expect(getByTestId('link')).toHaveTextContent('Danger By Design');
        expect(getByTestId('about')).toHaveTextContent('About Me');
        //assert 
    })
})