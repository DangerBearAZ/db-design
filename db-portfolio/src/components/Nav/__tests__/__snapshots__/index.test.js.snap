// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="flex-row px-1"
  >
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        <span
          aria-label="skull"
          role="img"
        >
          ðŸ’€
        </span>
         Danger By Design
      </a>
    </h2>
    <nav>
      <ul
        class="felx-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About Me
          </a>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Projects
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Resume
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
